megatron/core/inference/contexts/dynamic_context.py

```diff
@@ -682,15 +683,26 @@ def from_config(
         """
         # TODO: Add other necessary configs from inference_config
 
-        # Max sequence length. (todo: why does inference config have default value
-        # for inference_max_seq_length?)
-        model_max_seq_len = model.get_max_sequence_length()
+        # Max sequence length.
+        def get_model_attr(key):
+            try:
+                return getattr(model, key)
+            except Exception as e:
+                raise e
+
+        position_embedding_type = get_model_attr("position_embedding_type")
+        model_max_seq_len = get_model_attr("max_sequence_length")
         inf_max_seq_len = inference_config.inference_max_seq_length
-        if inf_max_seq_len:
-            max_sequence_length = min(model_max_seq_len, inf_max_seq_len)
+        if position_embedding_type == "learned_absolute":
```

Comment:
Can you add a comment explaining why we're only doing this check for `learned_absolute`?